hypertrace-view-generator/hypertrace-view-generator/src/main/java/org/hypertrace/viewgenerator/generators/RawServiceViewGenerator.java

```diff
@@ -29,7 +29,8 @@
     List<RawServiceView> list = new ArrayList<>();
 
     // Construct ApiTraceGraph and look at all the head spans within each ApiNode
-    ApiTraceGraph apiTraceGraph = new ApiTraceGraph(structuredTrace);
+    ApiTraceGraph apiTraceGraph = ViewGeneratorState.getApiTraceGraph(structuredTrace);
```

Comment:
I think it will be easy not to make functions static, and use ViewGeneratorState as a singleton class. It would really simplify writing the unit tests for the class

Reply:
I will give it a try, and if it works will make the changes in a subsequent pr.

Reply:
If it is made a singleton, remember that it should be stateless :)